internal/cmd/hana/check.go

```diff
@@ -155,25 +155,46 @@ func getServiceBinding(config *hanaCheckConfig, name string) (*unstructured.Unst
 		Get(config.ctx, name, metav1.GetOptions{})
 }
 
-func isReady(u *unstructured.Unstructured) (bool, error) {
+func getServiceStatus(u *unstructured.Unstructured) (Status, error) {
 	instance := somethingWithStatus{}
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &instance); err != nil {
-		return false, &clierror.Error{
+		return Status{}, &clierror.Error{
 			Message: "failed to read resource data",
 			Details: err.Error(),
 		}
 	}
-	status := instance.Status
+
+	return instance.Status, nil
+}
+
+// isReady returns readiness status, and failed status if at least one of the contitions has failed, or an error was returned
+func isReady(u *unstructured.Unstructured) (bool, bool, error) {
```

Comment:
Why this function return 3 values? IMO we have mixed two different functions `isReady` and `isFailed` and I don't see pros of mixing them together.